### DISCUSSION TITLE
Model for XML files (.nuspec)

### DISCUSSION THREAD

**Discussion by Starmania** [chocolatey-community/chocolatey-packages#2185] [Ideas]
### Before starting, ensure that you have done the following.

- [X] I have verified that there is no other issue or discussion covering the feature or enhancement
- [X] I have verified there is no open pull request for the feature or enhancement
- [X] I have verified the package to improve is located in this repository

### What do you want to add?

I suggest adding a new scheme to the XML .nuspec file, which would provide a standardized template for developers to create NuGet packages with ease. Currently, there is no such template available, and the Microsoft template provided is not compatible with some use cases. This creates a problem for developers who have to manually create the file from scratch, leading to errors and inconsistencies.

### What problem will the feature/enhancement solve?

The addition of a new scheme to the XML .nuspec file would solve these problems by providing a standardized format that aligns with the Microsoft format used by NuGet. This would make it easier for developers to create packages as they would have a clear idea of the necessary arguments and parameters needed to populate the fields in the XML file. Additionally, it would reduce the potential for errors and inconsistencies, ensuring that packages are created correctly and efficiently. By providing a standardized template for the XML .nuspec file, developers would be able to streamline their package creation process, leading to improved productivity and better quality packages.

### If the feature or enhancement is approved:

- [X] I am willing to create a pull request to implement the feature or enhancement

### Additional Context

- https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/compiler/resources/nuspec.xsd

**Comment by TheCakeIsNaOH** [comment]
> I suggest adding a new scheme to the XML .nuspec file, which would provide a standardized template for developers to create NuGet packages with ease.

Chocolatey packages are not the same as NuGet packages. They have different `.nuspec` schemas.
I'm not sure why Chocolatey would create a model/template for NuGet packages.

There is already an existing template for Chocolatey packages, which can be generated with the `choco new` command.

**Comment by Starmania** [comment]
The choco create does not include non-mandatory argumente like path.
The schemas btw NuGet and Choco are really similar, that why I'm referring to it...

Creating a schema takes only a few seconds, create more compatibility with existing code editors and why not take advantage of it to create compatibility between different existing systems (new suggestion)

**Reply by TheCakeIsNaOH** [reply]
> The choco create does not include non-mandatory argumente like path.

I'm not sure what you are referring to. I think all of the nuspec elements that are normally used in Chocolatey packages are included in the default template.

> Creating a schema takes only a few seconds, create more compatibility with existing code editors and why not take advantage of it to create compatibility between different existing systems (new suggestion)

Talking about schemas, that `.nuspec` schema from NuGet.Client you linked is actually not an accurate schema for what is allowed in a `.nuspec` by NuGet. Just off the top of my head, the `license` and `licenseUrl` elements are not allowed at the same time in a `.nuspec` by `nuget.exe`, and there are element length limits when uploading packages to NuGet.org.

Chocolatey v1.x does have a similar xml schema here: https://github.com/chocolatey/nuget-chocolatey/blob/2.11_adds/src/Core/Authoring/nuspec.xsd
And for the upcoming 2.0.0 release, there is a xml schema here: https://github.com/chocolatey/NuGet.Client/blob/develop/src/NuGet.Core/NuGet.Packaging/compiler/resources/nuspec.xsd

Like NuGet, Chocolatey CLI has additional restrictions on what is allowed vs the xml schemas, so they cannot just be used directly for accurate information.

I do think that more documentation for Chocolatey `.nuspec` files is required, there is an issue open here: https://github.com/chocolatey/docs/issues/374

**Reply by Starmania** [reply]
I agree that a better documentation of the .nuspec file would be better. 


> And for the upcoming 2.0.0 release, there is a xml schema here: https://github.com/chocolatey/NuGet.Client/blob/develop/src/NuGet.Core/NuGet.Packaging/compiler/resources/nuspec.xsd

If there is the schema file, why is it not linked to it by default?
